### PR TITLE
[SPARK-14884][SQL][Streaming][WebUI] Fix call site for continuous queries

### PR DESCRIPTION
**Pull request by lw-lin**
## What changes were proposed in this pull request?

Since we've been processing continuous queries in separate threads, the call sites are then `run at <unknown>:0`. It's not wrong but provides very little information; in addition, we can not distinguish two queries only from their call sites. 

This patch fixes this.
### Before

[Jobs Tab]
s1a
[SQL Tab]
s1b
### After

[Jobs Tab]
s2a
[SQL Tab]
s2b
## How was this patch tested?

Manually checks - see screenshots above.

### CODE REVIEW

**Comment by lw-lin** [comment]
@andrewor14 @zsxwing would you mind taking a look? Thanks! :-)

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala [hunk 1]
```diff
@@ -99,7 +99,15 @@ class StreamExecution(
   /** The thread that runs the micro-batches of this stream. */
   private[sql] val microBatchThread =
     new UninterruptibleThread(s"stream execution thread for $name") {
-      override def run(): Unit = { runBatches() }
+
+      private val callSite = Utils.getCallSite()
```
Comment (zsxwing) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala]: I prefer to make this as a field of `StreamExecution` as it's more intuitive.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala [hunk 2]
```diff
@@ -46,7 +45,11 @@ private[sql] object SQLExecution {
       val executionId = SQLExecution.nextExecutionId
       sc.setLocalProperty(EXECUTION_ID_KEY, executionId.toString)
       val r = try {
-        val callSite = Utils.getCallSite()
+        // We first try to pick up any call site that was set previously, then fall back to
+        // Utils.getCallSite(); because call Utils.getCallSite() on continuous queries directly
+        // would give us call site like "run at <unknown>:0"
+        val callSite = sparkSession.sparkContext.getCallSite()
```
Comment (zsxwing) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala]: Where is the fall back logic?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala (hunk 1 above)
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala]: @zsxwing making it as a field of `StreamExecution` is surely more reasonable; I've updated this within the new commit. Thanks!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala (hunk 2 above)
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala]: @zsxwing sorry for the vague comments. The fall back logic is contained in the `sparkSession.sparkContext.getCallSite()` itself. I've updated the comments accordingly.

**Comment by lw-lin** [comment]
@zsxwing I've made updates per your comments; would you take a another look? Thanks!

**Comment by zsxwing** [comment]
LGTM. Merging to master and 2.0, thanks!

**Pull request closed without merging**